src/renderers/webgl/WebGLExtensions.js

```diff
@@ -41,7 +53,26 @@ function WebGLExtensions( gl ) {
 					break;
 
 				default:
-					extension = gl.getExtension( name );
+
+					if ( isWebGL2 &&
+						[ 'ANGLE_instanced_arrays',
+						  'OES_texture_float',
+						  'OES_texture_half_float',
+						  'OES_texture_half_float_linear',
+						  'OES_element_index_uint',
+						  'OES_standard_derivatives',
+						  'EXT_frag_depth',
+						  'EXT_shader_texture_lod',
+						  'EXT_blend_minmax',
+						  'WEBGL_draw_buffers' ].indexOf( name ) >= 0 ) {
+
+						extension = gl;
+
+					} else {
+
+						extension = gl.getExtension( name );
+
+					}
```

Comment:
Returning `gl` for WebGL2.0 built-in features has Pros and Cons. Pros is less existing code change. Cons is a bit misleading.

For example, Three.js devs would think as if WebGL 2.0 also has ANGLE_instanced_arrays extension although it's built-in feature in WebGL 2.0 with this code.

	var extension = extensions.get( 'ANGLE_instanced_arrays' );

	extension[ isWebGL2 ? 'vertexAttribDivisor' : 'vertexAttribDivisorANGLE' ]( attribute, 0 );
	attributeDivisors[ attribute ] = 0;

Reply:
I made branch, removing workaround from `WebGLExtensions` and checking if context is webgl2 instead.

https://github.com/mrdoob/three.js/compare/dev...takahirox:WebGLRendererWebGL2.0_2

The code will be

    if ( isWebGL2 ) {

        gl.something();

    } else {

        var extension = extensions.get( 'something' );
        extension.something_EXT();

    }

Change will be a bit bigger, but less misleading. Personally I prefer this.

Reply:
I agree that returning `gl` on extensions feels strange. But I also don't like the second proposal that much, very verbose probably.
I know that I just wrote before that it's not a good idea to pollute the WebGLRenderContext, but the approach babylon is following is truly cleaner for the code: https://github.com/BabylonJS/Babylon.js/blob/ecbba4cd58b945732f10a474b897287cc322d14e/src/Engine/babylon.engine.ts#L1594
Maybe some mix approach where we save the reference to the extension or gl2 call to a variable on WebGLRenderer and use it directly? Not sure I'll think more about it

Reply:
In any case I don't see it as a blocking issue, we could merge it and later decide the best way to fix this.

Reply:
Agree with brushing up later. Personally I prefer starting with the second one because I like less misleading and probably we couldn't remove all `if ( isWebGL2 ) {} else {}` blocks.

Reply:
Yep :+1:

Reply:
Updated